packages/@react-aria/table/src/useTableRow.ts

```diff
@@ -34,9 +47,32 @@ export function useTableRow<T>(props: GridRowProps<T>, state: TableState<T> | Tr
     delete rowProps['aria-rowindex'];
   }
 
-  let treeGridRowProps = {};
+  let treeGridRowProps: HTMLAttributes<HTMLElement> = {};
   if ('expandedKeys' in state && state.collection.getItem(node.key)) {
     treeGridRowProps = {
+      onKeyDown: (e) => {
+        if ((e.key === EXPANSION_TRIGGERS['expand'][direction]) && state.selectionManager.focusedKey === node.key && node.props.childItems && state.expandedKeys !== 'all' && !state.expandedKeys.has(node.key)) {
+          state.toggleKey(node.key);
+          e.stopPropagation();
+          // After some time, if the row was not expanded (via controlled), focus the first cell
+          setTimeout(() => {
+            if (state.expandedKeys !== 'all' && !state.expandedKeys.has(node.key)) {
+              let cells = [...getChildNodes(node, state.collection)].filter(child => child.type === 'cell');
+              state.selectionManager.setFocusedKey(direction === 'ltr' ? cells[0].key : cells[cells.length - 1].key);
+            }
+          }, 10);
```

Comment:
this timeout feels a bit specific for a case that probably won't happen(?) since users would modify their controlled expandedKeys on toggle, I think I can go ahead and change the tests to use defaultExpandedKeys instead of controlled expandedKeys. We can ask for opinions from the team.

Reply:
in testing, the behavior diverges from aria grid examples since the row expands and then focus is moved to the next cell. Felt kinda weird to me IMO, open to opinions